mlflow/transformers/__init__.py

```diff
@@ -1858,6 +1888,11 @@ def _predict(self, data):
 
         data = self._convert_cast_lists_from_np_back_to_list(data)
 
+        # if applicable, wrap the data in the prompt template that was saved
+        # with the model. this function is a no-op if no prompt template was
+        # specified, or if the pipelines/inputs are not compatible with templating.
+        data = self._wrap_input_in_prompt_template(data)
```

Comment:
```suggestion
        # wrap the data in the prompt template that was saved with the model
        if self.prompt_template and self.pipeline.task in _SUPPORTED_PROMPT_TEMPLATING_TASK_TYPES:
             data = self._wrap_input_in_prompt_template(data)
```
nit: I think this is more self-explanatory than no-op?